packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQueryLoader.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ExistingQueryEditorStore } from '@finos/legend-application-query';
+import { clsx } from '@finos/legend-art';
+import type { QueryBuilderState } from '@finos/legend-query-builder';
+import { debounce, guaranteeNonNullable } from '@finos/legend-shared';
+import { flowResult } from 'mobx';
+import { observer } from 'mobx-react-lite';
+import { useState, useMemo } from 'react';
+import { DataSpaceQueryBuilderState } from '../../stores/query/DataSpaceQueryBuilderState.js';
+import {
+  createQueryDataSpaceTaggedValue,
+  createQueryClassTaggedValue,
+} from '../../stores/query/DataSpaceQueryCreatorStore.js';
+
+export const DataSpaceQueryLoaderFilterOptions = observer(
```

Comment:
I think for something like this, the action is very very deliberate and restrictive, should the filter mechanism be also structured in that same way instead of going full-customized like this?

Reply:
what are you suggesting here?

Reply:
I mean, instead of letting people have a whole component for the action in the extension, we can restrict to an action config: the name of the filter, the function that the filter call to get the results, etc. can things be constructed in that way?